js/rollup.config.mjs

```diff
@@ -0,0 +1,78 @@
+import typescript from '@rollup/plugin-typescript';
+import resolve from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
+import dts from 'rollup-plugin-dts';
+import json from '@rollup/plugin-json'; // Import the @rollup/plugin-json to handle JSON files
+
+const generateBundleAndTypeBundles = (file) => {
+
+  return [{
+    input: `src/${file}.ts`,
+    output: [
+      // CommonJS bundle
+      {
+        file: `dist/${file}.js`,
+        format: 'cjs',
+        sourcemap: true
+      },
+    ],
+    plugins: [
+      resolve(), // Resolve node_modules
+      commonjs(), // Convert CommonJS modules to ES6
+      typescript({
+        tsconfig: './tsconfig.json',
+        compilerOptions: { 
+          declaration: false, // Handled separately by dts plugin
+          module: 'esnext' // Ensure TypeScript produces ES Modules
+        }
+      }),
+      json() // Add the json plugin to handle JSON files
+    ],
+    external: [
```

Comment:
The external dependencies list is empty but should be populated to avoid bundling node_modules. Consider adding common external dependencies:

```javascript
external: [
  // Node.js built-ins
  'path',
  'fs',
  'util',
  // Third-party dependencies
  'axios',
  'winston',
  'zod',
  // All node_modules
  /node_modules/
]
```

This helps reduce bundle size and avoid duplicate dependencies.